# Libraries

In [ ]:
library(tidyverse)

In [ ]:
library(data.table)

In [ ]:
install.packages("faraway", quiet = TRUE)

In [ ]:
library(faraway)

# Implementation

# F3

## F3.2.a

To find which of the predictors are significant we can straightforwardly use the `lm()` model.

In [ ]:
cheddar_linear_model <- lm(taste ~ Acetic + H2S + Lactic, data = cheddar)
head(cheddar)
summary(cheddar_linear_model)

Per the summary, the model is significantly better than an intercept only null model. 

For the question of which predictors are significant, H2S and Lactic acid are significant at the 0.05 level.

## F3.2.b

To that end, you can use `exp()` to remove the natural log for the predictor values.

In [ ]:
# `chedar_log_removed` is a place holder. It will have the log-transformation removed in the next step.

cheddar_log_removed <- cheddar

In [ ]:
cheddar_log_removed[,c("Acetic","H2S","Lactic") ]<- exp(cheddar_log_removed[,c("Acetic","H2S","Lactic")])

In [ ]:
cheddar_log_removed_linear_model <- lm(taste ~ Acetic + H2S + Lactic, data = cheddar_log_removed)

In [ ]:
summary(cheddar_log_removed_linear_model)

Per the summary of the model with the log scale removed, the model itself is still significantly better than the null model or one of the independent variables does reduce the mean square error.

Without the log scale, only the Lactic acid content is a significant predictor of the taste score.

## F3.2.c

This question has two parts.

#### Can the two models be compared using an F-test?

Short answer: No.

We can use linear model's F-statistic to compare a linear model with predictors against the null model that has the intercept only. Extending that idea, we can use the *partial F-test* to compare models where one model is simple, linear and nested subset of the other model that is "complex" but still linear. For our answer, we want to compare a model that is on the log scale to a model that is on the original scale. The log scale is not a nested linear transformation of the original scale. As such we cannot use the partial F-test to compare the two models.

#### Which model provides a better fit to the data?

Short answer: The model on the log scale.

The log-scale model has a R-squared of 0.6518 where the original scale model has a R-squared of 0.5794.

## F3.2.d

The Estimated Regression Coefficnet for H2S is 3.9118. So, for every unit increase in H2S, the taste score is expected to increase by 3.9118. Which means for an increase of 0.01 in H2S, the taste score is expected to increase by 0.01*3.9118 = 0.039118

## F3.2.e

Note from the student: Note sure if the the expectation was to do this in R or if plan pseducode is enough.

This should be a percent incerease of `exp(.01)`

In [ ]:
f32e_answer = ((exp(.01) - 1 ) * 100)

In [ ]:
print(f32e_answer)

So, the value is 1.005017 %.

# F 3.4

## F3.4.a

In [ ]:
sat_linear_model <- lm(total ~ expend + ratio + salary, data = sat)

In [ ]:
summary(sat_linear_model)

#### Hypothesis: &beta;Salary = 0

While the coefficient is non-zero, it is not significant at the 5% level. So, we fail to reject the null hypothesis.

#### Hypothesis: &beta;Salary = &beta;Ratio = &beta;Expend = 0

The F-test is significant at the 5% level. So, we reject the null hypothesis i.e. the model is a better fit than the null model that used only the intercept without any predictors.

## F3.2.b

Testing the model with takes included.

In [ ]:
sat_takers_linear_model <- lm(total ~ expend + ratio + salary + takers, data = sat)

In [ ]:
summary(sat_takers_linear_model)

#### Hypothesis: &beta;takers = 0
 
The coefficient for &beta;takers with -2.9045 is significant at the 5% level. So, we reject the null hypothesis.

#### Compare the model to the previous model

In [ ]:
anova(sat_linear_model, sat_takers_linear_model)

#### Simple vs. Complex Model
 
Per the `anova()` output, the complex model does explain more variance than the simple model.

#### Is the partial F-test a T-test?

The only difference between the simple model and the complex model is that of a single predictor i.e. "the takers". In effect, the partial F-test has asked the question does "Does the takers as a predictor explain a significant amount of variance in the dependent variable?".

So, witihin this context, the partial F-test is a T-test and also F = t^2.

# F3.5

```Typst
#set page(width: auto, height: auto, margin: 1cm)
#set text(font: "Noto Sans", size: 11pt)

= Relationship: $R^2$ and the $F$-statistic

In the context of an Ordinary Least Squares (OLS) regression with $n$ observations and $k$ independent variables (excluding the intercept), the $F$-statistic for the global null hypothesis $H_0: beta_1 = beta_2 = dots = beta_k = 0$ is expressed as a function of the coefficient of determination $R^2$:

$F = frac(R^2 / k, (1 - R^2) / (n - k - 1))$

== Variables

$*n*$: Total number of observations.

$*k*$: Number of predictors (parameters excluding the intercept).

$*R^2*$: Coefficient of determination ($SS_"reg" / SS_"tot"$).

$*n - k - 1*$: Degrees of freedom for the error term.

== Derivation
The general definition of the $F$-statistic is the ratio of Mean Square Regression ($MS_"reg"$) to Mean Square Error ($MS_"err"$):

$F = frac(SS_"reg" / k, SS_"err" / (n - k - 1))$

Given the identity $R^2 = frac(SS_"reg", SS_"tot")$ and $1 - R^2 = frac(SS_"err", SS_"tot")$, we substitute:

$F = frac((R^2 dot SS_"tot") / k, ((1 - R^2) dot SS_"tot") / (n - k - 1))$

The $SS_"tot"$ terms cancel, yielding the $R^2$ form.

== Special Case: Simple Linear Regression ($k=1$)
For a single predictor, the formula simplifies to:

$F = frac(R^2(n - 2), 1 - R^2) = t^2$

where $t$ is the t-statistic for the slope coefficient.
```

# F3.6

In [ ]:
data(happy)

In [ ]:
head(happy)

## F3.6.a

In [ ]:
happy_linear_model <- lm(happy ~ money + sex + love + work, data = happy)

# change the significance level to 1%
confint(happy_linear_model, level = 0.99)

In [ ]:
summary(happy_linear_model)

#### Which predictors are significant at the 1% level?

In [ ]:
summary(happy_linear_model)$coefficients[, "Pr(>|t|)"] < 0.01

Seems the only predictor that is significant at the 1% level is Love.

## F3.6.b

In [ ]:
table(happy$happy)

Seems the data is not normally distributed.

In [ ]:
shapiro.test(happy$happy)

We can confirm with the `shapiro.test()` function that the data is not normally distributed.

## F3.6.c

Here is the code that I used for this.

In [ ]:
# Setup
set.seed(42)
B <- 500000
obs_model <- lm(happy ~ money + sex + love + work, data = happy)
t_obs <- summary(obs_model)$coefficients["money", "t value"]

# Permutation Loop
t_null <- replicate(B, {
  happy$money_perm <- sample(happy$money)
  perm_model <- lm(happy ~ money_perm + sex + love + work, data = happy)
  summary(perm_model)$coefficients["money_perm", "t value"]
})

# Two-tailed p-value calculation
p_perm <- mean(abs(t_null) >= abs(t_obs))

# Visualization
hist(t_null, breaks = 50, main = "Null Distribution of t-values",
     xlab = "Permuted t-statistics", xlim = range(c(t_null, t_obs)))
abline(v = t_obs, col = "red", lwd = 2, lty = 2)

cat("Observed t:", t_obs, "\n")
cat("Permutation p-value (B =", B, "):", p_perm, "\n")

Per the statistics from the permutation test, the p-value is 0.07. As such we fail to reject the null hypothesis. The money predictor does not have a significant effect on the depedence variable (happyness).

## F3.6.d

Histogram for the permutation of the t-statistic.

In [ ]:
hist(t_null, freq = FALSE)
grid = seq(-3,3,length=300)
lines(grid,dt(grid,df=34))

## F3.6.e

Note from the student: Is this not what we did for F3.6.d?

Anyways, here is how I would do this.

In [ ]:
# Histogram on density scale
hist(t_null, breaks = 50, freq = FALSE, 
     main = "Null Distribution vs. Theoretical t-distribution",
     xlab = "t-statistic", ylim = c(0, 0.4))

# Theoretical t-density grid
grid <- seq(-4, 4, length = 300)
# Use 34 degrees of freedom from the original lm output
t_density <- dt(grid, df = 34)

# Superimpose
lines(grid, t_density, col = "blue", lwd = 2)
abline(v = t_obs, col = "red", lwd = 2, lty = 2)

legend("topright", legend = c("Permuted Dist", "Theoretical t(34)"),
       col = c("gray", "blue"), lwd = 2)

## F3.6.f

In [ ]:
library(boot)

# Function to return the money coefficient
boot_fn <- function(data, indices) {
  model <- lm(happy ~ money + sex + love + work, data = data[indices, ])
  return(coef(model)["money"])
}

# Execute bootstrap
boot_results <- boot(data = happy, statistic = boot_fn, R = 2000)

# Calculate intervals (Normal, Basic, Percentile, BCa) for 90% and 95%
boot.ci(boot_results, conf = c(0.90, 0.95), type = c("norm", "basic", "perc", "bca"))

#### Does the confidence interval include 0?
Per the `boot()` function in R and the `*BCA*` based bootstrap it looks like 0 is not included in either the 90% and 95% confidence intervals.

#### What does the confidence interval say about the previous tests?

The Linear Model predict the coefficient to be 0.009578 but at a significance of 0.07 against a threshold of 0.05. Where the Boostrap has predicted the intervals to be significant at a threshold of 0.05 or 5% with a consistently postiitive impact.

# F3.7

In [ ]:
punting_linear_model <- lm(Distance ~ RStr + LStr + RFlex + LFlex, data = punting)

In [ ]:
summary(punting_linear_model)

## F3.7.a

Seems like none of the predictors are significant at the 0.05 threshold.

## F3.7.b

Per the `lm()` model, the F-statistic is 5.59 and is significant at the 0.05 level.

# F3.7.c

I used `linearHypothesis` function 

In [ ]:
install.packages("car")

In [ ]:
library(car)

In [20]:
linearHypothesis(punting_linear_model, "RStr = LStr")

Res.Df RSS      Df Sum of Sq F         Pr(>F)   
1 9      2287.356 NA       NA         NA        NA
2 8      2132.641  1 154.7151  0.5803702 0.4680292

With a P-value of 0.468, we fail to reject the null hypothesis i.e. the Right-strenght and the Left-strength are equal.

## F3.7.d

In [ ]:
# Plot the 95% confidence ellipse for the two strength coefficients
confidenceEllipse(punting_linear_model, which.coeff = c("RStr", "LStr"), levels = 0.95)

Looks like the confidence intervals do agree with the assesment from C with `(0,0)` being in the 95% CI.


## F3.7.e

In [ ]:
# Reduced Model (Testing if Total Strength is sufficient)
punting_total_strenght_linear_model <- lm(Distance ~ I(RStr + LStr) + RFlex + LFlex, data = punting)

# Partial F-test
anova(punting_total_strenght_linear_model, punting_linear_model)

Our null hypothesis is that total Strength is the same as using a granular right strength, left strenght. Given that out P-value is not less than 0.05, we fail to reject the null hypothesis. So, seems we can just use the total strength.

## F3.7.f

I used the `car` package's `linearHypothesis()` function to perform the linear hypothesis test.

In [21]:
linearHypothesis(punting_linear_model, "RFlex = LFlex")

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 9      2648.358 NA       NA        NA        NA
2 8      2132.641  1 515.7173  1.934568 0.2017235

With a P-value of 0.201we fail to reject the null hypothesis i.e. the Right-flex the Left-flex are equal in terms of effect.

## F3.2.g

In [24]:
# Full Model: 4 predictors
fm <- lm(Distance ~ RStr + LStr + RFlex + LFlex, data = punting)

# Symmetry Model: 2 predictors (Total Strength and Total Flexibility)
# This model assumes side-to-side symmetry
sm <- lm(Distance ~ I(RStr + LStr) + I(RFlex + LFlex), data = punting)

# Partial F-test for Symmetry
anova(sm, fm)

Res.Df RSS      Df Sum of Sq F        Pr(>F)   
1 10     2799.075 NA       NA        NA        NA
2  8     2132.641  2 666.4339  1.249969 0.3369868

For the null hypothesis that Right and Left parts of the system have similar impact, we have a P-value of 0.336. It seems like the null hypothesis is not rejected and the two parts of the system have similar impact.

## F3.2.h

In [26]:
punting_hang_linear_model <- lm(Hang ~ RStr + LStr + RFlex + LFlex, data = punting)

In [27]:
anova(punting_linear_model, punting_hang_linear_model)

Warning message in anova.lmlist(object, ...):
“models with response ‘"Hang"’ removed because response differs from model 1”


Df Sum Sq     Mean Sq    F value    Pr(>F)     
RStr      1  5069.80896 5069.80896 19.0179581 0.002409544
LStr      1    50.42669   50.42669  0.1891615 0.675109657
RFlex     1   731.48721  731.48721  2.7439679 0.136213140
LFlex     1   108.94490  108.94490  0.4086761 0.540510899
Residuals 8  2132.64071  266.58009         NA          NA

As the warning message states, we can but should not compare the two models. The predictors are shared but the dependent is not. As such, comparision in not valid.

# Scratch

In [ ]:
data(punting)

In [23]:
head(punting)

Distance Hang RStr LStr RFlex LFlex OStr  
1 162.50   4.75 170  170  106   106   240.57
2 144.00   4.07 140  130   92    93   195.49
3 147.50   4.04 180  170   93    78   152.99
4 163.50   4.18 160  160  103    93   197.09
5 192.00   4.35 170  150  104    93   266.56
6 171.75   4.16 150  150  101    87   260.56